File: app/controllers/Application.java
package controllers;

import controllers.core.BaseController;
import model.Secured;
import play.Environment;
import play.data.FormFactory;
import play.mvc.Result;
import play.mvc.Security;

import javax.inject.Inject;

@Security.Authenticated(Secured.class) public class Application extends BaseController {

  private Environment environment;

  @Inject public Application(FormFactory f, Environment theEnvironment) {
    super(f);
    environment = theEnvironment;
  }

  public Result index() {
    return ok(views.html.index.render(getUser(), environment.isDev()));
  }

}

File: app/controllers/Application.scala
package controllers

import javax.inject.Inject
import model.core.Repository
import model.toolMains.ToolList
import play.api.db.slick.{DatabaseConfigProvider, HasDatabaseConfigProvider}
import play.api.mvc.{AbstractController, ControllerComponents, EssentialAction}
import slick.jdbc.JdbcProfile

import scala.concurrent.ExecutionContext

class Application @Inject()(cc: ControllerComponents, val dbConfigProvider: DatabaseConfigProvider, toolList: ToolList, val repository: Repository)(implicit ec: ExecutionContext)
  extends AbstractController(cc) with HasDatabaseConfigProvider[JdbcProfile] with Secured with play.api.i18n.I18nSupport {

  override protected val adminRightsRequired: Boolean = false

  def index: EssentialAction = withUser { user =>
    implicit request => Ok(views.html.index(user, toolList))
  }

}
